### PR TITLE
Cura 7427 add option to sign in with different account while waiting for printers

### PR DESCRIPTION
**Pull request by konskarm**
Adds a button in the "Waiting for cloud response" page, which will allow the user to force logout from Cura and from mycloud.ultimaker.com in the browser, in order to login again and sync the accounts between Cura and the browser.

### CODE REVIEW

File: cura/API/Account.py [hunk 1]
```diff
@@ -96,6 +96,18 @@ def login(self) -> None:
             return
         self._authorization_service.startAuthorizationFlow()
 
+    @pyqtSlot()
+    def loginWithForcedLogout(self) -> None:
```
Comment (ninovanhooff) [inline cura/API/Account.py:100]: maybe add a parameter forceLogout to the login function instead?

File: tests/TestOAuth2.py [hunk 2]
```diff
@@ -226,3 +226,19 @@ def test_wrongServerResponses() -> None:
     with patch.object(AuthorizationHelpers, "parseJWT", return_value=UserProfile()):
         authorization_service._onAuthStateChanged(MALFORMED_AUTH_RESPONSE)
     assert authorization_service.getUserProfile() is None
+
+
+def test__generate_auth_url() -> None:
```
Comment (ninovanhooff) [inline tests/TestOAuth2.py:231]: tests. good.

File: cura/API/Account.py (hunk 1 above)
Comment (konskarm) [reply cura/API/Account.py]: Fixed with https://github.com/Ultimaker/Cura/pull/7743/commits/eac4d3e463c659f9f18ca3a3d7aff603f2561933. Tests have also been adjusted accordingly.

File: cura/API/Account.py [hunk 3]
```diff
@@ -90,11 +90,23 @@ def _onLoginStateChanged(self, logged_in: bool = False, error_message: Optional[
...
     @pyqtSlot()
-    def login(self) -> None:
+    @pyqtSlot(bool)
+    def login(self, force_logout_before_login: bool = False) -> None:
+        """
+        Initializes the login process. If the user is logged in already and force_logout_before_login is true, Cura will
+        logout from the account before initiating the authorization flow. If the user is logged in and
+        force_logout_before_login is false, the function will return, as there is nothing to do.
+
+        :param force_logout_before_login: Optional boolean parameter
```
Comment (ninovanhooff) [inline cura/API/Account.py:100]: This doesn't add any information beyond what is visible from the signature. I would leave it out next time

**Pull request merged**